requirements-dev.txt

```diff
@@ -1,10 +1,28 @@
-# necessary dev/test tasks
-pylint
-coverage
-diff_cover
+# Used by tox jobs to install test suite dependencies.
+# CI dependencies are only required in the specific tox environments.
+# Maybe move these as development deps to pyproject.toml later:
```

Comment:
These can be done with:

```toml
[project.optional-dependencies]
test = [
    "mock",
    "pytest",
    "pytest-cov",
    "typing_extensions"
]
```
Then on tox.ini:
```
-deps        = -r requirements-dev.txt
+extras      = test
```
See [extras](https://tox.wiki/en/4.5.1/config.html#extras)

After that, the all requirements*.txt should be able to be deleted

Reply:
Done!
- Works nicely with Python3.7 which supports tox version 4
- Needed to use Python3.8 to run tox version 4 and specify to run the 3.6 and 2.7 tests.